src/Aspire.Cli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,307 @@
+            {
+                _interactionService.DisplayError(ExecCommandStrings.FailedToParseCommand);
+                return ExitCodeConstants.InvalidCommand;
+            }
+
+            string[] args = [
+                "--operation", "exec",
+                targetResourceMode, targetResource!,
+
+                // a bit hacky, but in order to pass a full command with possible quotes and etc properly without losing the signature
+                // we can wrap it in a string and pass it as a single argument
+                "--command", $"\"{string.Join(" ", commandTokens)}\"",
+
+                ..arbitraryFlags,
+            ];
+
+            var backchannelCompletionSource = new TaskCompletionSource<IAppHostBackchannel>();
+            var pendingRun = _runner.RunAsync(
+                projectFile: effectiveAppHostProjectFile,
+                watch: watch,
+                noBuild: !watch,
+                args: args,
+                env: env,
+                backchannelCompletionSource: backchannelCompletionSource,
+                options: runOptions,
+                cancellationToken: cancellationToken);
+
+            // We wait for the back channel to be created to signal that
+            // the AppHost is ready to accept requests.
+            var backchannel = await _interactionService.ShowStatusAsync(
+                $":linked_paperclips:  {RunCommandStrings.StartingAppHost}",
+                async () =>
+                {
+                    // If we use the --wait-for-debugger option we print out the process ID
+                    // of the apphost so that the user can attach to it.
+                    if (waitForDebugger)
+                    {
+                        _interactionService.DisplayMessage(emoji: "bug", InteractionServiceStrings.WaitingForDebuggerToAttachToAppHost);
+                    }
+
+                    // The wait for the debugger in the apphost is done inside the CreateBuilder(...) method
+                    // before the backchannel is created, therefore waiting on the backchannel is a
+                    // good signal that the debugger was attached (or timed out).
+                    var backchannel = await backchannelCompletionSource.Task.WaitAsync(cancellationToken);
+                    return backchannel;
+                });
+
+            _ = await _interactionService.ShowStatusAsync<int>(
+                ":running_shoe: Running exec...",
+                async () =>
+                {
+                    // execute tool and stream the output
+                    var outputStream = backchannel.ExecAsync(cancellationToken);
+                    await foreach (var output in outputStream)
+                    {
+                        _interactionService.WriteConsoleLog(output.Text, output.LineNumber, output.Type, output.IsErrorMessage);
+                    }
+
+                    return ExitCodeConstants.Success;
+                });
+
+            if (!keepAliveAppHost)
```

Comment:
This may not work the way you expect. The app host as a CLI orphan detector so that if the CLI crashes, the apphost will detect it and shut itself down. This is a protection against orphan processes.